src/main/java/cs2103/v15_1j/jimjim/command/DeleteCommand.java

```diff
@@ -28,7 +28,12 @@ public char getPrefix() {
     @Override
     public UIFeedback undo(ControllerStates conStates) {
 		// Add task/event back at former position
-	    conStates.masterList.add(taskNum-1, backup);
+	    conStates.masterList.add(backup);
+	    
+	    if(!conStates.searchResultsList.isEmpty()){
```

Comment:
@SorataYuu ideally everytime masterList is updated, searcher should see whether the change affect searchResultLists and update it if needed. I think the Observer pattern can be used here. But for this pr, just remove these adding back is sufficient.